llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -1243,8 +1243,12 @@ splitSYCLModule(std::unique_ptr<Module> M, ModuleSplitterSettings Settings) {
 }
 
 bool canBeImportedFunction(const Function &F) {
+  // SYCL_EXTERNAL property is not recorded for a declaration
+  // in a header file.  Thus SYCL IR that is a declaration
+  // will be considered as SYCL_EXTERNAL for the purposes of
+  // this function.
   if (F.isIntrinsic() || F.getName().starts_with("__") ||
-      !llvm::sycl::utils::isSYCLExternalFunction(&F))
+      !(F.isDeclaration() || llvm::sycl::utils::isSYCLExternalFunction(&F)))
```

Comment:
nit: can we distribute the `!`, so like

`(!F.isDeclaration() && !llvm::sycl::utils::isSYCLExternalFunction(&F))`

it's easier to read IMO

Reply:
Updated expression.